src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -2,56 +2,45 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.IO.Pipelines;
 using System.Net.Http;
 using System.Threading.Tasks;
+using System.Threading.Tasks.Channels;
+using Microsoft.AspNetCore.Sockets.Internal;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Sockets.Client
 {
-    public class Connection : IPipelineConnection
+    public class Connection : IChannelConnection<Message>
```

Comment:
I've been thinking about this a bit and I wonder if there's an upside in exposing the channel interfaces on the client side. The benefit in theory is the ability to share code between client and server but we aren't really doing that. We should make sure the design is right here before going too far with it.